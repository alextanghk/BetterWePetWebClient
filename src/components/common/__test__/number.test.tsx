import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { BWPNumber } from '..';
import renderer from 'react-test-renderer'

afterEach(cleanup)


describe("Vertical BWP Number Render checking", ()=>{
    test('Vertical BWP Number rendering correctly in md size', () => {
        const { getByTestId } = render(<BWPNumber vertical={true} textAlign="left" inputSize="md" />)
        const testItem = getByTestId("bwp-number")
        const testInput = getByTestId("bwp-number-input")
        expect(testItem).toBeTruthy()
    })

    test("snapshort of vertical bwp number in md size", () => {
        const snap = renderer.create(<BWPNumber vertical={true} textAlign="left" inputSize="md" />).toJSON()
        expect(snap).toMatchSnapshot()
    })
})

describe("Horizontal BWP Number Render checking", ()=>{
    test('Horizontal BWP Number rendering correctly in md size', () => {
        const { getByTestId } = render(<BWPNumber vertical={false} textAlign="left" inputSize="md" defaultValue={0}/>)
        const testItem = getByTestId("bwp-number")
        const testInput = getByTestId("bwp-number-input") as HTMLInputElement
        expect(testItem).toBeTruthy()
        expect(testInput.value).toEqual("0")
    })
    test("snapshort of horizontal bwp number in md size", () => {
        const snap = renderer.create(<BWPNumber vertical={false} textAlign="left" inputSize="md" />).toJSON()
        expect(snap).toMatchSnapshot()
    })
})

describe("BWP Number Function Check",()=>{
    test("When add value",() => {
        const { getByTestId } = render(<BWPNumber vertical={false} textAlign="left" inputSize="md" defaultValue={0}/>)
        const testClick = getByTestId("bwp-number-increase")
        const testInput = getByTestId("bwp-number-input") as HTMLInputElement
        fireEvent.click(testClick)
        expect(testInput.value).toEqual("1")
    })

    test("When add value with max limited",() => {
        const { getByTestId } = render(<BWPNumber vertical={false} textAlign="left" inputSize="md" defaultValue={3} max={3}/>)
        const testClick = getByTestId("bwp-number-increase")
        const testInput = getByTestId("bwp-number-input") as HTMLInputElement
        fireEvent.click(testClick)
        expect(testInput.value).toEqual("3")
    })

    test("When decrease value",() => {
        const { getByTestId } = render(<BWPNumber vertical={false} textAlign="left" inputSize="md" defaultValue={0}/>)
        const testClick = getByTestId("bwp-number-decrease")
        const testInput = getByTestId("bwp-number-input") as HTMLInputElement
        fireEvent.click(testClick)
        expect(testInput.value).toEqual("-1")
    })

    test("When decrease value with min limited",() => {
        const { getByTestId } = render(<BWPNumber vertical={false} textAlign="left" inputSize="md" defaultValue={1} min={1}/>)
        const testClick = getByTestId("bwp-number-decrease")
        const testInput = getByTestId("bwp-number-input") as HTMLInputElement
        fireEvent.click(testClick)
        expect(testInput.value).toEqual("1")
    })
})

