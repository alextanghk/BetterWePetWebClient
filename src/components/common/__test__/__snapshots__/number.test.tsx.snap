// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal BWP Number Render checking snapshort of horizontal bwp number in md size 1`] = `
<div
  className="undefined"
>
  <div
    className=" relative flex w-full p-0.5 box-content bg-bwp-grey h-6 rounded-6"
    data-testid="bwp-number"
  >
    <img
      className="h-full flex-initial cursor-pointer"
      data-testid="bwp-number-increase"
      onClick={[Function]}
      src="ico_plus.svg"
    />
    <input
      className="bg-bwp-grey text-left w-full text-base"
      data-testid="bwp-number-input"
      type="number"
    />
    <img
      className="[object Object] cursor-pointer"
      data-testid="bwp-number-decrease"
      onClick={[Function]}
      src="ico_minus.svg"
    />
  </div>
</div>
`;

exports[`Vertical BWP Number Render checking snapshort of vertical bwp number in md size 1`] = `
<div
  className="undefined"
>
  <div
    className=" relative flex flex-col p-0.5 box-content bg-bwp-grey w-6 rounded-6"
    data-testid="bwp-number"
  >
    <img
      className="w-full cursor-pointer"
      data-testid="bwp-number-increase"
      onClick={[Function]}
      src="ico_plus.svg"
    />
    <input
      className="bg-bwp-grey text-left w-full text-base"
      data-testid="bwp-number-input"
      type="number"
    />
    <img
      className="[object Object] cursor-pointer"
      data-testid="bwp-number-decrease"
      onClick={[Function]}
      src="ico_minus.svg"
    />
  </div>
</div>
`;
